### PR TITLE
move nlp request to effect

### PR DESCRIPTION
**Pull request by fdore**

### CODE REVIEW

File: src/client/src/apps/SpotlightRoute/Spotlight.tsx [hunk 1]
```diff
@@ -62,6 +69,37 @@ export const Spotlight: FC = () => {
...
+        (err: any) => {
+          console.error(err)
+          dispatch({type: 'RECEIVE_RESPONSE', response: null})
+        },
+      )
+
+    return () => {
+      subscription.unsubscribe()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (sizov) [inline src/client/src/apps/SpotlightRoute/Spotlight.tsx]: why did you need this comment? was it because you didn't want to add `request` as dep?

Comment (fdore) [reply src/client/src/apps/SpotlightRoute/Spotlight.tsx]: because request keeps changing as you type. And I only want to trigger the request when contacting goes to `true`

Comment (fdore) [reply src/client/src/apps/SpotlightRoute/Spotlight.tsx]: because we have this 
```
const handleChange: ChangeEventHandler<HTMLInputElement> = e => {
    dispatch({type: 'SET_REQUEST', request: e.target.value})
  }
```

Comment (sizov) [reply src/client/src/apps/SpotlightRoute/Spotlight.tsx]: that's interesting, I first thought that then `request` should probablt go into ref (not the state), but for now maybe it would be worth just adding a comment on line 99 :

>> because request keeps changing as you type. And I only want to trigger the call when contacting goes to true
